devices/YarpRobotLoggerDevice/src/YarpRobotLoggerDevice.cpp

```diff
@@ -1239,6 +1249,226 @@ void YarpRobotLoggerDevice::recordVideo(const std::string& cameraName, VideoWrit
+
+    // configure the metadata for the inertail measurments
+    if (m_streamInertials)
+    {
+        for (const auto& sensorName: m_robotSensorBridge->getGyroscopesList())
+        {
+            std::string fullGyroSensorName = "robot_realtime::gyros::" + sensorName;
+            vectorCollectionRTDataServer.populateMetadata(fullGyroSensorName, {"omega_x", "omega_y", "omega_z"});
+        }
+        for (const auto& sensorName: m_robotSensorBridge->getLinearAccelerometersList())
+        {
+            std::string fullAccelerometerSensorName = "robot_realtime::accelerometers::" + sensorName;
+            vectorCollectionRTDataServer.populateMetadata(fullAccelerometerSensorName, {"a_x", "a_y", "a_z"});
+        }
+        for (const auto& sensorName: m_robotSensorBridge->getOrientationSensorsList())
+        {
+            std::string fullOrientationSensorName = "robot_realtime::orientations::" + sensorName;
+            vectorCollectionRTDataServer.populateMetadata(fullOrientationSensorName, {"r", "p", "y"});
+        }
+        for (const auto& sensorName: m_robotSensorBridge->getMagnetometersList())
+        {
+            std::string fullMagnetometersSensorName = "robot_realtime::magnetometers::" + sensorName;
+            vectorCollectionRTDataServer.populateMetadata(fullMagnetometersSensorName, {"mag_x", "mag_y", "mag_z"});
+        }
+    }
+
+    if(m_streamTemperatureSensors)
+    {
+        for (const auto& sensorName: m_robotSensorBridge->getTemperatureSensorsList())
+        {
+            std::string fullTemperatureSensorName = "robot_realtime::temperature::" + sensorName;
+            vectorCollectionRTDataServer.populateMetadata(fullTemperatureSensorName, {"temperature"});
+        }
+    }
+
+    if (m_streamCartesianWrenches)
+    {
+        for (const auto& cartesianWrenchName : m_robotSensorBridge->getCartesianWrenchesList())
+        {
+            std::string fullCartesianWrenchName = "robot_realtime::cartesian_wrench::" + cartesianWrenchName;
+            vectorCollectionRTDataServer.populateMetadata(fullCartesianWrenchName, {"f_x", "f_y", "f_z", "mu_x", "mu_y", "mu_z"});
+        }
+    }
+
+    for (auto& [name, signal] : m_vectorsCollectionSignals)
+    {
+        externalSignalCollection = signal.client.readData(false);
+        if (externalSignalCollection != nullptr)
+        {
+            for (const auto& [key, vector] : externalSignalCollection->vectors)
+            {
+                std::string signalFullName = "robot_realtime::" + signal.signalName + "::" + key;
+                vectorCollectionRTDataServer.populateMetadata(signalFullName, {});
+            }
+        }
+    }
+
+    vectorCollectionRTDataServer.finalizeMetadata();
+}
+
+void YarpRobotLoggerDevice::SendDataToLoggerVisualizer()
+{
+    const double time = std::chrono::duration<double>(BipedalLocomotion::clock().now()).count();
+    Eigen::Matrix<double, 1, 1> timeData;
+    timeData << time;
+
+    vectorCollectionRTDataServer.clearData();
+
```

Comment:
@GiulioRomualdi  I think that a ``prepareData`` call will be needed when https://github.com/ami-iit/bipedal-locomotion-framework/pull/790 gets merged